util/regex_safety.py

```
@@ -0,0 +1,160 @@
+        "." * 50 + "!",
+        "_" * 50 + "!",
+        "a1" * 25 + "!",
+        "a.b" * 17 + "!",
+    ]
+
+    for test_input in adversarial_inputs:
+        result: list[bool] = []
+
+        def do_match():
+            try:
+                compiled.match(test_input)
```

Ruff (0.15.0)

[warning] Function definition does not bind loop variable `test_input`

(B023)